File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/commandbase/SleepCommand.java
package org.firstinspires.ftc.teamcode.common.commandbase;


import androidx.annotation.NonNull;

import com.qualcomm.robotcore.util.ElapsedTime;

/**
 * Usage:
 * Waits a certain amount of time before returning true
 */
public class SleepCommand extends Command {
    private ElapsedTime timer;
    private final double timeoutSeconds;

    /**
     * Waits a certain amount of time before returning true
     * @param commandName The name of the command
     * @param timeoutSeconds The amount of time to wait
     */
    public SleepCommand(String commandName, double timeoutSeconds) {
        super(commandName);
        this.timeoutSeconds = timeoutSeconds;
    }

    /**
     * Waits a certain amount of time before returning true
     * @param timeoutSeconds The amount of time to wait
     */
    public SleepCommand(double timeoutSeconds) {
        this.timeoutSeconds = timeoutSeconds;
    }

    @Override
    public boolean run() {
        if(timer == null)
            timer = new ElapsedTime();

        //TODO: test if this works
        return timer.seconds() >= timeoutSeconds;
    }

    @NonNull
    @Override
    public Command clone() {
        return new SleepCommand(getCommandName(), timeoutSeconds);
    }
}

File: TeamCode/src/main/java/brickbot/quickstart/commandbase/SleepCommand.kt
package brickbot.quickstart.commandbase

import com.qualcomm.robotcore.util.ElapsedTime

class SleepCommand(
    commandName: String = "",
    private val timeoutMs: Long
): Command(commandName) {
    private lateinit var timer: ElapsedTime

    override fun run(): Boolean {
        if (!::timer.isInitialized) {
            timer = ElapsedTime()
            timer.reset()
        }

        return timer.milliseconds() > timeoutMs
    }

    override fun clone(): Command {
        return SleepCommand(commandName, timeoutMs)
    }
}